airbyte-commons/src/main/java/io/airbyte/commons/json/Jsons.java

```diff
@@ -220,25 +220,28 @@ public static int getIntOrZero(final JsonNode json, final List<String> keys) {
   }
 
   /**
-   * Flattens an ObjectNode, or dumps it into a {null: value} map if it's not an object.
+   * Flattens an ObjectNode, or dumps it into a {null: value} map if it's not an object. When
+   * applyFlattenToArray is true, each element in the array will be one entry in the returned map.
+   * This behavior is used in the Redshift SUPER type. When it is false, the whole array will be one
+   * entry. This is used in the JobTracker.
    */
-  public static Map<String, Object> flatten(final JsonNode node) {
+  public static Map<String, Object> flatten(final JsonNode node, final boolean applyFlattenToArray) {
     if (node.isObject()) {
       final Map<String, Object> output = new HashMap<>();
       for (final Iterator<Entry<String, JsonNode>> it = node.fields(); it.hasNext();) {
         final Entry<String, JsonNode> entry = it.next();
         final String field = entry.getKey();
         final JsonNode value = entry.getValue();
-        mergeMaps(output, field, flatten(value));
+        mergeMaps(output, field, flatten(value, false));
```

Comment:
@marcosmarxm this cannot be hardcoded to `false`, but needs to pass `applyFlattenToArray` through - you can have an array nested inside an object.